DataMigration/src/main/java/org/sagebionetworks/dian/datamigration/BridgeJavaSdkUtil.java

```diff
@@ -0,0 +1,314 @@
+        if (user.externalId.equals(user.arcId)) {
+            return false;
+        }
+        String isMigratedStr = participant.getAttributes().get(ATTRIBUTE_IS_MIGRATED);
+        return ATTRIBUTE_VALUE_TRUE.equals(isMigratedStr);
+    }
+
+    /**
+     * Create user attributes for signing up a user to Bridge.
+     * @param user the user that will be signed up on Bridge.
+     * @return the user attributes to attach to the SignUp object.
+     */
+    protected static Map<String, String> newUserAttributes(HmDataModel.HmUser user) {
+        String raterEmail = NO_RATER_ASSIGNED_YET_EMAIL;
+        if (user.rater != null && user.rater.email != null) {
+            raterEmail = user.rater.email;
+        }
+        String notes = "";
+        if (user.notes != null) {
+            notes = user.notes;
+        }
+        String phoneNum = "";
+        if (user.phone != null) {
+            phoneNum = user.phone;
+        }
+
+        Map<String, String> attributeMap = new HashMap<>();
+        attributeMap.put(ATTRIBUTE_IS_MIGRATED, ATTRIBUTE_VALUE_FALSE);
+        attributeMap.put(ATTRIBUTE_ARC_ID, user.arcId);
+        attributeMap.put(ATTRIBUTE_RATER_EMAIL, raterEmail);
+        attributeMap.put(ATTRIBUTE_SITE_NOTES, notes);
+        attributeMap.put(ATTRIBUTE_PHONE_NUM, phoneNum);
+
+        // This is a critical part of the user management system for the DIAN project.
+        // A user's password is stored securely as an attribute for study managers to see.
+        // They will use this to sign their study participants in.
+        // User attributes are encrypted at rest and during transit.
+        attributeMap.put(ATTRIBUTE_VERIFICATION_CODE, user.password);
+
+        // Make sure the attributes are supported by Bridge
+        bridgifyAttributes(attributeMap);
+
+        return attributeMap;
+    }
+
+    /**
+     * Bridge requires attributes to be less than a certain length
+     * This function truncates attributes that are too long so that they save to bridge.
+     * @param attributes to make sure they will upload to bridge
+     */
+    protected static void bridgifyAttributes(Map<String, String> attributes) {
+        for (String key: attributes.keySet()) {
+            String val = attributes.get(key);
+            attributes.put(key, val.substring(0, Math.min(val.length(), ATTRIBUTE_LENGTH_MAX)));
+        }
+    }
+
+    /**
+     * Make a "singleton" report as we call it, where the report is at Jan. 1st, 1970.
+     * Sage mobile apps use this to store information that should only be stored once.
+     * @param clientData to attach to the report
+     * @return a report that can be uploaded to bridge.
+     */
+    private static ReportData makeReportData(String clientData) {
+        ReportData reportData = new ReportData();
+        reportData.setLocalDate(REPORT_DATE);
+        reportData.setData(clientData);
+        return reportData;
+    }
```

Comment:
I was very tempted to switch over to that, but at this point it would be too much work and too risky to update the Android/iOS apps to use that instead; however, this is definitely going to be my approach for future projects.